src/ch05-03-method-syntax.md

```diff
@@ -33,62 +33,62 @@ fn main() {
-name, parentheses, and any arguments.
-
-In the signature for `area`, we use `&self` instead of `rectangle: &Rectangle`
-because Rust knows the type of `self` is `Rectangle` due to this method’s being
-inside the `impl Rectangle` context. Note that we still need to use the `&`
-before `self`, just as we did in `&Rectangle`. Methods can take ownership of
-`self`, borrow `self` immutably as we’ve done here, or borrow `self` mutably,
-just as they can any other parameter.
-
-We’ve chosen `&self` here for the same reason we used `&Rectangle` in the
-function version: we don’t want to take ownership, and we just want to read the
-data in the struct, not write to it. If we wanted to change the instance that
-we’ve called the method on as part of what the method does, we’d use `&mut
-self` as the first parameter. Having a method that takes ownership of the
-instance by using just `self` as the first parameter is rare; this technique is
-usually used when the method transforms `self` into something else and you want
-to prevent the caller from using the original instance after the transformation.
-
-The main benefit of using methods instead of functions, in addition to using
-method syntax and not having to repeat the type of `self` in every method’s
-signature, is for organization. We’ve put all the things we can do with an
-instance of a type in one `impl` block rather than making future users of our
-code search for capabilities of `Rectangle` in various places in the library we
-provide.
-
-> ### Where’s the `->` Operator?
+<span class="caption">Listing 5-13: Definicja metody `area` w strukturze `Rectangle`</span>
+
+Aby zdefiniować funkcję w kontekście `Rectangle` otwieramy blok implementacji `impl`. 
+Następnie przenosimy funkcję `area` do nawiasów klamrowych
+należących do bloku `impl` i jako pierwszy parametr funkcji dodajemy
+`self` w jej sygnaturze i wszędzie wewnątrz funkcji. W funkcji `main`,
+zamiast jak poprzednio wywołania funkcji `area`, a jako argument
+podawania jej `rect1`, możemy użyć *składni metody* aby wywołać metodę `area`
+na instancji struktury `Rectangle`.
+Składnia metody pojawia się po nazwie instancji: dodajemy kropkę, a po niej nazwę samej metody, oraz
+nawiasy i argumenty jeśli są wymagane.
+
+W sygnaturze funkcji `area` używamy `&self` zamiast `rectangle: &Rectangle`
+ponieważ Rust wie, że typ `self` to `Rectangle`, gdyż metoda znajduje się
+wewnątrz kontekstu `impl Rectangle`. Zauważ, że nadal musimy użyć `&`
+przed `self`, podobnie jak w `&Rectangle`. Metody mogą wejść w posiadanie `self`,
+pożyczyć `self` bez możliwości zmiany, lub pożyczyć `self` z możliwością zmiany, 
+tak jakby to był jakikolwiek inny parametr.
+
+W tym wypadku użyliśmy `&self` z tego samego powodu, co `&Rectangle`
+w wersji z funkcją. Otóż nie chcemy zostać właścicielem, ale nadal powinniśmy móc odczytać dane
+ze struktury, nie potrzebujemy jednak jej zmieniać. Jeśli chcielibyśmy zmienić dane instancji w trakcie wywoływania
```

Comment:
`ale nadal powinniśmy móc odczytać dane
ze struktury, nie potrzebujemy jednak jej zmieniać.`

sugestie bo brzmi kiepsko?

Reply:
Może: "...w wersji z funkcją: nie chcemy zostać właścicielem struktury, ponieważ nie chcemy do niej pisać, a jedynie z niej czytać."

Reply:
Mnie się osobiście to pisanie i czytanie się za bardzo nie klei po polsku. Jak @beling nie widzi problemu to mogę zmienić

Reply:
Trzymając się zgodnie z oryginałem, raczej powinno być:
nie chcemy ani zostać właścicielem struktury, ani do niej pisać, a jedynie z niej czytać.

Reply:
zmienione